lua/randomat2/events/sv_poker.lua

```diff
@@ -0,0 +1,1136 @@
+--// Logan Christianson
+
+util.AddNetworkString("StartPokerRandomat")
+util.AddNetworkString("StartPokerRandomatCallback")
+util.AddNetworkString("BeginPokerRandomat")
+util.AddNetworkString("NotifyBlinds")
+util.AddNetworkString("DealCards")
+util.AddNetworkString("StartBetting")
+util.AddNetworkString("MakeBet")
+util.AddNetworkString("PlayerFolded")
+util.AddNetworkString("PlayerChecked")
+util.AddNetworkString("PlayerCalled")
+util.AddNetworkString("PlayerRaised")
+util.AddNetworkString("StartDiscard")
+util.AddNetworkString("PlayersFinishedBetting")
+util.AddNetworkString("MakeDiscard")
+util.AddNetworkString("DeclareNoWinner")
+util.AddNetworkString("DeclareWinner")
+util.AddNetworkString("ClosePokerWindow")
+
+util.AddNetworkString("ShareCards")
+util.AddNetworkString("MarkRoundVariant")
+
+--// EVENT properties
+
+local EVENT = {}
+local EVENT_VARIANT = {}
+local EVENT_REF = nil
+
+EVENT.Title = "A Round Of Yogscast Poker"
+EVENT.Description = "Only if the 9 of Diamonds touch!"
+EVENT.ExtDescription = "A round of 5-Card Draw Poker (no Texas Hold 'Em, for my sake), bet with your\nhealth. Up to 7 may play. Any pair, three, or four of a kind containing the 9 of\nDiamonds instantly wins."
+EVENT.id = "poker"
+EVENT.MinPlayers = 2
+EVENT.Type = EVENT_TYPE_DEFAULT
+EVENT.Categories = {"gamemode", "largeimpact", "fun"}
+
+--// My properties
+
+EVENT.NumberOfGames = 0
+EVENT.MaxPlayers = 7
+EVENT.Started = false
+EVENT.Running = false
+EVENT.Players = {}
+EVENT.ContinuousPlayers = {}
+EVENT.Deck = {}
+EVENT.PlayerBets = {}
+
+--// EVENT Functions
+
+-- Used to populate EVENT.Players with living players, up to the max amount
+function EVENT:GeneratePlayers()
+    local removedPlayers = {}
+    local playersToPlay = {}
+
+    -- If continuous play is enabled, return the previous player list, clean up any disconnected players, and add in as many new players as possible
+    if ConVars.EnableContinuousPlay:GetBool() and #self.ContinuousPlayers > 0 then
+        playersToPlay = table.Copy(self.ContinuousPlayers)
+
+        local playersToRemove = {}
+        for _, ply in ipairs(playersToPlay) do
+            if !IsValid(ply) or !ply:IsValid() or not ply:Alive() then
```

Comment:
Consistency. Also "IsValid" already calls `ply:IsValid`
```suggestion
            if not IsValid(ply) or not ply:Alive() then
```